File: C#/Practice/LeetCode/Easy/Height Checker.cs
// https://leetcode.com/problems/height-checker/description/

public class Solution {
    public int HeightChecker(int[] heights) {
        int[] sortedHeights = heights.ToArray();
        Array.Sort(sortedHeights);
        int notMatching = 0;
        
        for (int i = 0; i < heights.Length; i++) {
            if (heights[i] != sortedHeights[i]) {
                notMatching++;
            }
        }
        
        return notMatching; 
    }
}

File: TypeScript/Practice/LeetCode/Easy/Height Checker.ts
// https://leetcode.com/problems/height-checker/description/

function heightChecker(heights: number[]): number {
    const sortedHeights: number[] = [...heights].sort((a, b) => a - b);
    let notMatching: number = 0;
    
    for (let i = 0; i < heights.length; i++) {
        if (heights[i] !== sortedHeights[i]) {
            notMatching++;
        }
    }
    
    return notMatching;    
};